README.md

```diff
@@ -1,29 +1,286 @@
+
+<!--
+  - [name=Stef] or do we just define what workflow they have to use? Could also give a comma separated list of branch names in the --git-service option?
+-->
+
+You can omit `--service github` if the repo is at `github.com` (as in
+this case) and you can omit `--repo <url>` if you have the desired
+repository configured as the git remote named `upstream`.
+
+#### Setup Personal Access Token
+
+- Follow the [github instructions](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-fine-grained-personal-access-token) we suggest the following configuration options:
+
+  | field             | value, user defined values defined by `${...}`              |
+  | ----------------- | ----------------------------------------------------------- |
+  | Token name        | `qw-${repository_name}`                                     |
+  | Expiration        | `custom...` then choose 364 days from today                 |
+  | Description       | `QW tool for ${repository_name}`                            |
+  | Repository access | `Only select repositories` -> Select the project repository |
+  | Permissions       | Only add repository permissions, see next table             |
+
+  | repository permissions | value          |
+  | ---------------------- | -------------- |
+  | Actions                | Read and write |
+  | Contents               | Read-only      |
+  | Issues                 | Read and write |
+  | Metadata               | Read-only      |
+  | Pull Requests          | Read and write |
+
+  Then click the `Generate token` button and copy the token (you can't get its value again if you close the page).
+
+- Add the personal access token to your machine's keychain (uses [keyring](https://pypi.org/project/keyring/))
+
+  ```shell
+  qw login
+  ```
+
+  <!--
+     - [name=Stef] For this I guess we should implement it in keyring using: service=`qw:github.com` username=`owner/repo`?
+     - [name=Stef] Should try and get a windows user early on to test that keyring works as expected
+  -->
+
+  > Please copy the personal access token and hit enter:
+
+  After you've done this, the system will check the personal access token. If you've configured for the remote URL <https://github.com/stefpiatek/dummy-medical-software> you would see:
+
+  > INFO: Can access "stefpiatek/dummy-medical-software"
+
+### Setup for a new user, on an existing project that uses QW
+
+The local and github repository have already been configured to work with `qw`, you'll just need to [add the personal access token](#setup-personal-access-token)
+
+### Customising configuration
+
+QW initialises with a `System` component for design inputs (used for generating a unique identifier, short code value is `X`).
+You can add extra components to the configuration using qw:
+
+```shell
+qw config add-service --name="Drug dosage" --short-code=D
```

Comment:
Are we going to end up with a huge amount of different subcommands if we do it like this? This is one of those things I want to use MD for (except it's not clear we can sensibly). A big services.md document like:
```
### Service Name (SN)
This service names the service.
### Other Service Name (OS)
This service others other services.
```
Here I'm just using `()` to indicate the short code, which might not be sensible. `[SN](qw://short-code)` is more robust but also maybe horrible? Seems you are allowed links in headings.

Reply:
I guess if we're going down that route then maybe we should just ask them to write json/yaml and then validate it has the right fields and that they are unique?
```json
{
    {"name": "Service name", "short-code": "SN", "descirption": "This service names the service."},
    {"name": "Other service name", "short-code": "OSN", "descirption": "This service others other services."}
}
```